### ISSUE TITLE
[java] UnusedPrivateMethod false positive with static method and cast expression

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3209]
**Affects PMD Version:** 6.33.0

**Rule:** [UnusedPrivateMethod](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**
The usage of the static method is not detected anymore (not sure, if these details are relevant).
The false positive was not reported with PMD 6.32.0.
Maybe this is a side-effect of fixing #3118

**Code Sample demonstrating the issue:**

Method declaration: https://github.com/pmd/pmd/blob/30e841a9cadfbf6395857eedc987c6fdb3075a7d/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedAssignmentRule.java#L909

Method call: https://github.com/pmd/pmd/blob/30e841a9cadfbf6395857eedc987c6fdb3075a7d/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedAssignmentRule.java#L1396

**Expected outcome:**

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
Testcase:

```xml
    <test-code>
        <description>[java] UnusedPrivateMethod false positive with static method and cast expression #3209</description>
        <expected-problems>0</expected-problems>
        <code><![CDATA[
public class UnusedAssignmentRule {
    private static class ReachingDefsVisitor {
        private static String getVarIfUnaryAssignment(String node) {
            return null;
        }
    }
    public static class AssignmentEntry {
        final String var;
        final Object rhs;
        public boolean isUnaryReassign() {
            return rhs instanceof String
                && var.equals(ReachingDefsVisitor.getVarIfUnaryAssignment((String) rhs));
        }
    }
}
        ]]></code>
    </test-code>
```

**Comment by adangel** [comment]
It's fixed by #3683 as well
